Src/Couchbase.Linq/Extensions/BucketExtensions.cs

```diff
@@ -8,7 +9,8 @@ public static class BucketExtensions
     {
         public static IQueryable<T> Queryable<T>(this IBucket bucket)
         {
-            return EntityFilterManager.ApplyFilters(new BucketQueryable<T>(bucket));
+            //TODO refactor so ClientConfiguration is injectable
+            return EntityFilterManager.ApplyFilters(new BucketQueryable<T>(bucket, new ClientConfiguration()));
```

Comment:
I think BucketContext might make more sense than DbContext, and help avoid conflicts with EF in applications using both Couchbase and SQL/EF.

I certainly like removing the dependency on ClusterHelper from as much stuff as we can, especially within DbContext (or BucketContext) since it has a Configuration property.  And QueryFactory is internal.  But BucketExtensions is still a public API point.

If a user is getting an IBucket interface from ClusterHelper (instead of using DbContext), and then calls Queryable<T> on IBucket, I feel like their expectation will be that the serialization settings will be the one from the ClusterHelper.  But this implementation is currently not doing so in that use case.

Brant